x/voting/keeper/delegate_mint_burn.go

```diff
@@ -0,0 +1,103 @@
+package keeper
+
+import (
+	"encoding/hex"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/tendermint/tendermint/crypto"
+	"mande-chain/x/voting/types"
+)
+
+/*
+- Transfer `mand` tokens from caster account and lock it in module account
+- Mint calculated vote amount of `CRED` tokens to module account and invoke `stakingmodule.Delegate` to delegate to receiver (delegator should be voting module account)
+*/
+func (k Keeper) lockMandAndDelegateStake(ctx sdk.Context, msg *types.MsgCreateVote) error {
+	creator, _ := sdk.AccAddressFromBech32(msg.Creator)
+	voteCountAbs := int64(Abs(msg.Count))
+
+	// send tokens from the vote creator to the voting module account
+	voteCountEquivalentMand := sdk.Coins{sdk.NewInt64Coin("mand", voteCountAbs)}
+	err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, creator, types.ModuleName, voteCountEquivalentMand)
+	if err != nil {
+		return err
+	}
+
+	// mint stake token to voting module account
+	voteCountEquivalentStake := sdk.Coins{sdk.NewInt64Coin("stake", voteCountAbs)}
+	err = k.bankKeeper.MintCoins(ctx, types.ModuleName, voteCountEquivalentStake)
+	if err != nil {
+		return err
+	}
+
+	// delegate minted stake tokens to receiver
+	votingModuleAcct := sdk.AccAddress(crypto.AddressHash([]byte(types.ModuleName)))
+
+	receiverAccAddress, _ := sdk.AccAddressFromBech32(msg.Receiver)
+	receiverValAddr, err := sdk.ValAddressFromHex(hex.EncodeToString(receiverAccAddress.Bytes()))
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrReceiverIsNotAValidator, msg.Receiver)
+	}
+
+	recipientValidator, found := k.stakingKeeper.GetValidator(ctx, receiverValAddr)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrNoValidatorFound, receiverValAddr.String())
```

Comment:
Could not understand difference between this and error on line 40
Also should we move this validation here - https://github.com/mande-labs/mande-chain/blob/67139664e3566141136b12bebfd1050872e3dcdb/x/voting/types/message_create_vote.go#L42, might as well save some computation in case of error

Reply:
That error is to check if the receiver is a validator or not. That check needs to call keeper functions and is not a good idea to add in `ValidateBasic()`

Reply:
Added additional validations for `Count` and `Mode` in `ValidateBasic()`